TeamCode/src/main/java/org/firstinspires/ftc/teamcode/WaitNT.java

```diff
@@ -1,13 +1,30 @@
 package org.firstinspires.ftc.teamcode;
 
 public class WaitNT {
-
+    boolean completed = false;
+    double timeMS;
+    double targetMillisecondTime;
     /* Method that starts a timer that is not in while, returns true or false*/
-
+    public void waitAsync(double timeMS){
+        this.timeMS = timeMS;
+        targetMillisecondTime = timeMS + System.currentTimeMillis();
+    }
+    /** precondition: You must have called the waitSync method before this!**/
+    public boolean syncDone(){
+        return System.currentTimeMillis() > targetMillisecondTime; // parentheses are good practice here in order to emphasize that we do something with the returned boolean as opposed to an arbitrary return statement
+    }
     /* Method that starts a timer that is in a while, pauses all operations, (time out is 3 sec)  */
+    public void waitSync(double timeMS){
+        targetMillisecondTime = timeMS + System.currentTimeMillis();
+        this.timeMS = timeMS;
+        while (targetMillisecondTime > System.currentTimeMillis()) {
 
+        }
+    }
     /* Method that restarts the set timer (returns true false like first) */
-
+    public void resetTimer(){
+        targetMillisecondTime = timeMS + System.currentTimeMillis();
+    }
     /* Method that runs the while after starting the timer */
-    
+    //not implementing cause this seems the same as Method that starts a timer that is in a while, pauses all operations, (time out is 3 sec)
```

Comment:
This one should be similar to syncDone, After a timer has been started by waitAsync if this method is called, it should invoke a loop using that timer without creating a new one like you have made in waitSync